satpy/writers/cf_writer.py

```diff
@@ -767,9 +767,36 @@ def _collect_datasets(self, datasets, epoch=EPOCH, flatten_attrs=False, exclude_
 
         return datas, start_times, end_times
 
-    def save_datasets(self, datasets, filename=None, groups=None, header_attrs=None, engine=None, epoch=EPOCH,
-                      flatten_attrs=False, exclude_attrs=None, include_lonlats=True, pretty=False,
-                      compression=None, include_orig_name=True, numeric_name_prefix='CHANNEL_', **to_netcdf_kwargs):
+    def combine_info(self, group_name, group_datasets, epoch, group_attrs, flatten_attrs, exclude_attrs,
```

Comment:
Sorry, this got uglier than I was expecting. I think any sub-methods should be prefixed with `_` as they probably aren't public API for users. The main ugliness with this refactor is the number of arguments that needed to be passed. Looking at the arguments it looks like most of them go to the `_collect_datasets` call. Maybe that could be moved back into the for loop? Then the time if/else in one method and the group attrs in another method. So the for loop would essentially look like:

```python
for group_name, group_datasets ...:
    ... = self._collect_datasets(...)
    dataset = xr.Dataset(datas)
    self._add_time_bounds(...)
    self._add_group_attrs(...)
    encoding, other_to_netcdf_kwargs = update_encoding(...)
    return dataset, encoding, other_to_netcdf_kwargs
```